src/main_elevate.c

```diff
@@ -0,0 +1,67 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <unistd.h>
+#include <mach-o/dyld.h>
+#include <limits.h>
+#include <libgen.h>
+
+int main(int argc, char *argv[])
+{
+    char pathBuff[PATH_MAX];
+    uint32_t buffSize = PATH_MAX;
+
+    //
+    //  Find the absolute path
+    //  of this executable.
+    //
+    if (_NSGetExecutablePath(pathBuff, &buffSize) != KERN_SUCCESS)
+        return -1;
+
+    //
+    //  Remove the file component of the abs. path.
+    //
+    char *parentDir = dirname(pathBuff);
+
+    char *str;
+    int len = snprintf(
+        NULL, 
+        0, 
+        "do shell script \"sudo %s/cslol-manager >/dev/null 2>&1 &\""
+        " with prompt \"CSLOL-Manager needs administrator privileges to restart.\""
+        " with administrator privileges"
+        " without altering line endings",
+        parentDir
+    );
+
+    //
+    //  Something went wrong trying
+    //  to obtain the necessary length of the string.
+    //
+    if (len <= 0)
+        return 1;
+
+    str = (char *)malloc(len + 1);
```

Comment:
Malloc and 2 printfs  seems unecesary.
Since we know that path is not going be longer than MAX_PATH we could format to sufficiently sized fixed buffer on stack?
We could probably optimize this step further by somehow reusing same buffer.
I leave this up to you if you want to figure it out.

Reply:
We could allocate it onto the stack, though I genuinely do think it's completely irrelevant, as there isn't that much overhead with using `malloc` in this case.

Reply:
In that case can we deduplicate format string.
Having 2 string literals that can go out of sync is bugprone.

Reply:
:+1: